pkg/workceptor/workceptor.go

```diff
@@ -322,6 +323,9 @@ func (w *Workceptor) monitorRemoteStatus(ctx context.Context, cancel context.Can
 			logger.Error("Error saving local status file: %s\n", err)
 			return
 		}
+		if unit.status.State == WorkStateFailed {
```

Comment:
this makes sure we stop monitoring if the cancel goes through on the remote node.

Reply:
Shouldn't we also stop monitoring the cancel if the status is WorkStateSucceeded?

Also, there's an `IsComplete` helper function, so you can do `if IsComplete(unit.status.State)` which returns true if it's failed or succeeded, and is perhaps a little more future-proof if we ever add more work states.